File: rust/src/easy/sum_of_multiples.rs
fn is_divisible_by_any_factor(num: &u32, factors: &[u32]) -> bool {
    factors.iter().any(|factor| match factor {
        0 => false,
        _ => num % factor == 0,
    })
}

pub fn sum_of_multiples(limit: u32, factors: &[u32]) -> u32 {
    (1..limit)
        .filter(|num| is_divisible_by_any_factor(num, factors))
        .sum()
}

File: elixir/lib/easy/sum_of_multiples.ex
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, factors) do
    Enum.reduce(1..(limit - 1), 0, fn num, acc ->
      multiple?(factors, num) && acc + num || acc
    end)
  end

  defp multiple?(factors, num) do
    Enum.any?(factors, fn factor ->
      case factor do
        0 -> false
        _ -> rem(num, factor) == 0
      end
    end)
  end
end
